app/src/main/java/de/westnordost/streetcomplete/screens/main/map/QuestPinsManager.kt

```diff
@@ -58,7 +63,11 @@ class QuestPinsManager(
 
     private val visibleQuestsListener = object : VisibleQuestsSource.Listener {
         override fun onUpdatedVisibleQuests(added: Collection<Quest>, removed: Collection<QuestKey>) {
-            viewLifecycleScope.launch { updateQuestPins(added, removed) }
+            val oldPinsUpdateJob = pinsUpdateJob
+            pinsUpdateJob = viewLifecycleScope.launch {
+                oldPinsUpdateJob?.join() // don't cancel, as updateQuestPins only updates existing data
+                updateQuestPins(added, removed)
```

Comment:
I don't get it, why not?

Reply:
When panning the map while uploading, canceling on `onUpdated` will cancel the job for loading data currently in view, and then only update some elements / quests. Data in view may _not_ be loaded

I noticed this while testing, and not seeing any quests in an area you know you have downloaded is really bad.

Reply:
But when it is cancelled, it means that it is cancelled because another tile rect comes into view, no? And that means that either the updates are not in view anymore or they are part of the data just fetched by bbox from the db anyway.

Reply:
> But when it is cancelled, it means that it is cancelled because another tile rect comes into view, no?

Only cancelling as part of `onNewTilesRect`.
But here (in `onUpdatedVisibleQuests`) there is no new `TilesRect` coming in view, only some quests should be added/removed.

Reply:
I see, okay